src/MongoDB.Driver/Encryption/DataKeyOptions.cs

```diff
@@ -0,0 +1,76 @@
+﻿/* Copyright 2019-present MongoDB Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System.Collections.Generic;
+using MongoDB.Bson;
+
+namespace MongoDB.Driver.Encryption
+{
+    /// <summary>
+    /// Options for creating a data key.
+    /// </summary>
+    public class DataKeyOptions
+    {
+        // private fields
+        private readonly BsonDocument _masterKey;
+        private readonly IReadOnlyList<string> _alternateKeyNames;
+
+        // constructors
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DataKeyOptions"/> class.
+        /// </summary>
+        /// <param name="alternateKeyNames">The alternate key names.</param>
+        /// <param name="masterKey">The master key.</param>
```

Comment:
A short description about the expected format of the BsonDocument might be useful